chessington/engine/pieces.py

```diff
@@ -121,25 +121,87 @@ class Bishop(Piece):
     """
 
     def get_available_moves(self, board):
-        return []
+        current_square = board.find_piece(self)
+        available_moves = self.get_possible_squares(board, current_square)
+        return available_moves
 
+    def get_possible_squares(self, board, current_square):
+        possible_squares = []
+        directions = [-1, -1], [1, 1], [-1, 1], [1, -1]
+        for direction in directions:
+            square = current_square
+            invalid_direction = False
+            while invalid_direction is False:
+                square = Square.at(square.row + direction[0], square.col + direction[1])
+                if board.is_inside_bounds(square) is True:
+                    if board.get_piece(square) is None:
+                        possible_squares.append(square)
```

Comment:
Seems like this check is pretty repetitive among the pieces, can we extract it in a separate function?